dialect/sql/sqlgraph/graph.go

```diff
@@ -512,8 +512,10 @@ func OrderByNeighborTerms(q *sql.Selector, s *Step, opts ...sql.OrderTerm) {
 	case s.ToEdgeOwner():
 		toT := build.Table(s.Edge.Table).Schema(s.Edge.Schema)
 		join = build.Select(toT.C(s.Edge.Columns[0])).
-			From(toT).
-			GroupBy(toT.C(s.Edge.Columns[0]))
+			From(toT)
+		if s.Edge.Rel != O2O {
+			join = join.GroupBy(toT.C(s.Edge.Columns[0]))
```

Comment:
That’s true if the edge is unique, not only o2o. 

Also, please add/update tests.

Reply:
@a8m Do you have ideas for tests? It looks like there is already a test that covers ordering by a unique edge field, but all the tests use sqlite, and therefore this specific test passed both before and after this change. This change is only to generate valid sql in other databases too, such as PostgreSQL.